src/test/java/com/google/sps/UtilityTest.java

```diff
@@ -115,4 +118,93 @@ public void nullValue() {
 
     Assert.assertEquals(actual, expected);
   }
+
+  /** 
+   *  Tests for getUserEmailsAsString.
+   *  These only work by having an active MySQL database created with mysql/create.sql
+   *  and then populating with mysql/testPopulate.sql.
+   */
+  @Test
+  public void normalSingleUserQuery() {
```

Comment:
A common style practice for naming java unit test functions it to include the method being tested, the test condition, and the expected outcome, so something like: getUserEmailsAsString_normalSingleUserQuery_Success()

Doing this makes it really explicitly clear what is being tested, how it's being tested, and what we expect to see. Consider updating your methods to also follow this pattern!

It's not technically part of the public github Java style guide, so I'll leave it up to you - but a good habit to have :)

Reply:
alright, thanks!